backend/pennmobile/admin.py

```diff
@@ -1,11 +1,20 @@
 # CUSTOM ADMIN SETTUP FOR PENN MOBILE
+from typing import Any, Dict, Optional, Type, TypeAlias
+
 from django.contrib import admin, messages
 from django.contrib.admin.apps import AdminConfig
+from django.db.models import Model
+from django.http import HttpRequest
 from django.urls import reverse
 from django.utils.html import format_html
 
 
-def add_post_poll_message(request, model):
+ModelType: TypeAlias = Type[Model]
+AdminContext: TypeAlias = Dict[str, Any]
+MessageText: TypeAlias = str
```

Comment:
Same here, ifl ModelType should be in some common library (or better, installed library someone else has already put together). AdminContext would also benefit from coming from a common library (maybe more generically as `Context` as this is a common Django idea, or again better from some outside library). MessageText seems overkill.